### PR TITLE
Integrate Serializers for completion

### PR DESCRIPTION
**Pull request by bergundy**
This is making up for some tech debt where I introduced Serializers to the SDK and didn't apply it for the completion code.
I need this now for integrating into the server repo.

### CODE REVIEW

File: nexus/completion_test.go [hunk 1]
```diff
@@ -40,10 +41,11 @@ func TestSuccessfulCompletion(t *testing.T) {
 	ctx, callbackURL, teardown := setupForCompletion(t, &successfulCompletionHandler{})
 	defer teardown()
 
-	request, err := NewCompletionHTTPRequest(ctx, callbackURL, &OperationCompletionSuccessful{
-		Header: http.Header{"foo": []string{"bar"}},
-		Body:   bytes.NewReader([]byte("success")),
-	})
+	completion, err := NewOperationCompletionSuccessful("content", OperationCompletionSuccesfulOptions{})
```
Comment (MichaelSnowden) [inline nexus/completion_test.go:44]: I think we ought to have a test case where the serializer is set.

File: nexus/completion.go [hunk 2]
```diff
@@ -115,11 +140,14 @@ type CompletionHandlerOptions struct {
 	// A stuctured logging handler.
 	// Defaults to slog.Default().
 	Logger *slog.Logger
+	// A [Serializer] to customize handler serialization behavior.
+	// By default the handler handles, JSONables, byte slices, and nil.
+	Serializer Serializer
```
Comment (MichaelSnowden) [inline nexus/completion.go:145]: I'd be interested in knowing what the default serializer does when the handler returns something not in this list.

**Review by MichaelSnowden — APPROVED** [review]

File: nexus/completion_test.go (hunk 1 above)
Comment (bergundy) [reply nexus/completion_test.go:44]: I was too lazy to add it but I'll do it before merging.

File: nexus/completion.go (hunk 2 above)
Comment (bergundy) [reply nexus/completion.go:145]: It's very much like the default payload converter in temporal SDKs.
It will fail if it can't serialize to JSON.

**Pull request merged**